src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/MethodWithGCInfo.cs

```diff
@@ -271,7 +271,19 @@ protected override DependencyList ComputeNonRelocationBasedDependencies(NodeFact
 
             foreach (ISymbolNode node in _fixups)
```

Comment:
On the NativeAOT side we do this check as we compile the method - it ends up invalidating the whole body. Dropping the fixup here makes me uneasy but I don't know that much about R2R.

Reply:
I don't think we're actually dropping the fixup, we're just not using it for further dependency tree expansion i.o.w. methods only referred to via such fixup would not be precompiled and would require runtime JIT; please note this only pertains to the "pathologic" case of methods potentially causing generic cycles.

Reply:
This is dropping the dependencies of the fixup. I'm not very comfortable with that, and agree with @MichalStrehovsky here. Instead, could we detect the cycle when adding the fixup to _fixups, and if we do so, mark the method with whatever details we do, when then entire method can't be compiled.